### PR TITLE
GHA: ignore unrelated file changes

### PR DESCRIPTION
**Pull request by isinyaaa**
## Merge criteria:

- [ ] The commits and have meaningful messages; the author will squash them [after approval](https://github.com/opendatahub-io/opendatahub-community/blob/main/contributor-cheatsheet.md#:~:text=Usually%20this%20is%20done%20in%20last%20phase%20of%20a%20PR%20revision) or will ask to merge with squash.
- [ ] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [ ] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

File: .github/workflows/python-tests.yml [hunk 1]
```diff
@@ -64,9 +73,3 @@ jobs:
           else
             nox --python=${{ matrix.python }}
           fi
-      - name: Upload documentation
-        if: matrix.session == 'docs-build'
-        uses: actions/upload-artifact@v4
-        with:
-          name: docs
-          path: clients/python/docs/_build
```
Comment (tarilabs) [inline .github/workflows/python-tests.yml:72]: I'm not sure why this is removed, isn't this storing a preview of the docs while submitting PRs?

Comment (isinyaaa) [reply .github/workflows/python-tests.yml:72]: afaik there's currently no way to see these. if we enable PR docs build on readthedocs it still performs the build on their side

Comment (tarilabs) [reply .github/workflows/python-tests.yml:72]: > afaik there's currently no way to see these

go actions
Screenshot 2024-08-26 at 09 23 45

select gha workflow for python: 
Screenshot 2024-08-26 at 09 24 33

select a run, for instance here i selected latest from main default branch
https://github.com/kubeflow/model-registry/actions/runs/10510838464

Screenshot 2024-08-26 at 09 27 17

you obtain a zip,
and it's the preview of the docs when this gha job is run for the PRs instead.

do we want to remove this for some reasons?

Comment (isinyaaa) [reply .github/workflows/python-tests.yml:72]: oh I didn't know that, thanks for the pointer! I don't mind keeping it, I just thought it was kinda useless.

**Review by tarilabs — COMMENTED** [review]
/lgtm 

thank you 

/assign @ederign

**Review by ederign — APPROVED** [review]

File: .github/workflows/ui-bff-build.yml [hunk 2]
```diff
@@ -2,12 +2,13 @@ name: UI - BFF - Build
...
+      - "main"
   pull_request:
     paths-ignore:
-      - 'LICENSE*'
-      - 'DOCKERFILE*'
-      - '**.gitignore'
+      - "LICENSE*"
+      - "DOCKERFILE*"
+      - "**.gitignore"
+      - "**.md"
```
[the remark this answers is not in the document]
Comment (ederign) [reply .github/workflows/ui-bff-build.yml:11]: Seems good!

File: .github/workflows/ui-frontend-build.yml [hunk 3]
```diff
@@ -2,12 +2,13 @@ name: UI - Frontend - Test and Build
...
+      - "main"
   pull_request:
     paths-ignore:
-      - 'LICENSE*'
-      - 'DOCKERFILE*'
-      - '**.gitignore'
+      - "LICENSE*"
+      - "DOCKERFILE*"
+      - "**.gitignore"
+      - "**.md"
```
[the remark this answers is not in the document]
Comment (ederign) [reply .github/workflows/ui-frontend-build.yml:11]: Seems good!

**Review by tarilabs — APPROVED** [review]

**Pull request merged**